##### Задания не несут смысловой нагрузки для реальной жизни, они сделаны лишь для того, чтоб научить использовать Pandas

This database contains 76 attributes, but all published experiments refer to using a subset of 14 of them. In particular, the Cleveland database is the only one that has been used by ML researchers to this date. The "goal" field refers to the presence of heart disease in the patient. It is integer valued from 0 (no presence) to 1.

https://www.kaggle.com/ronitf/heart-disease-uci

___

In [1]:
import pandas as pd
import numpy as np

In [6]:
data = pd.read_csv("heart_pandas.csv")

In [37]:
# Хочу посмотреть первые пять строк - что в файле
data.head()

age     sex   chest_pain_type  resting_blood_pressure  cholesterol  \
0   63    male  non-anginal pain                     145          233   
1   37    male   atypical angina                     130          250   
2   41  female    typical angina                     130          204   
3   56    male    typical angina                     120          236   
4   57  female    typical angina                     120          354   

     fasting_blood_sugar               rest_ecg  max_heart_rate_achieved  \
0  greater than 120mg/ml                 normal                      150   
1    lower than 120mg/ml  ST-T wave abnormality                      187   
2    lower than 120mg/ml                 normal                      172   
3    lower than 120mg/ml  ST-T wave abnormality                      178   
4    lower than 120mg/ml  ST-T wave abnormality                      163   

  exercise_induced_angina  st_depression   st_slope  num_major_vessels  \
0                      no            2.3  upsloping                  0   
1                      no            3.5  upsloping                  0   
2                      no            1.4       flat                  0   
3                      no            0.8       flat                  0   
4                     yes            0.6       flat                  0   

    thalassemia  target  
0        normal       1  
1  fixed defect       1  
2  fixed defect       1  
3  fixed defect       1  
4  fixed defect       1

In [8]:
data.dtypes

age                          int64
sex                         object
chest_pain_type             object
resting_blood_pressure       int64
cholesterol                  int64
fasting_blood_sugar         object
rest_ecg                    object
max_heart_rate_achieved      int64
exercise_induced_angina     object
st_depression              float64
st_slope                    object
num_major_vessels            int64
thalassemia                 object
target                       int64
dtype: object

In [57]:
data

age     sex   chest_pain_type  resting_blood_pressure  cholesterol  \
0     63    male  non-anginal pain                     145          233   
1     37    male   atypical angina                     130          250   
2     41  female    typical angina                     130          204   
3     56    male    typical angina                     120          236   
4     57  female    typical angina                     120          354   
..   ...     ...               ...                     ...          ...   
298   57  female    typical angina                     140          241   
299   45    male  non-anginal pain                     110          264   
300   68    male    typical angina                     144          193   
301   57    male    typical angina                     130          131   
302   57  female    typical angina                     130          236   

       fasting_blood_sugar               rest_ecg  max_heart_rate_achieved  \
0    greater than 120mg/ml                 normal                      150   
1      lower than 120mg/ml  ST-T wave abnormality                      187   
2      lower than 120mg/ml                 normal                      172   
3      lower than 120mg/ml  ST-T wave abnormality                      178   
4      lower than 120mg/ml  ST-T wave abnormality                      163   
..                     ...                    ...                      ...   
298    lower than 120mg/ml  ST-T wave abnormality                      123   
299    lower than 120mg/ml  ST-T wave abnormality                      132   
300  greater than 120mg/ml  ST-T wave abnormality                      141   
301    lower than 120mg/ml  ST-T wave abnormality                      115   
302    lower than 120mg/ml                 normal                      174   

    exercise_induced_angina  st_depression   st_slope  num_major_vessels  \
0                        no            2.3  upsloping                  0   
1                        no            3.5  upsloping                  0   
2                        no            1.4       flat                  0   
3                        no            0.8       flat                  0   
4                       yes            0.6       flat                  0   
..                      ...            ...        ...                ...   
298                     yes            0.2  upsloping                  0   
299                      no            1.2  upsloping                  0   
300                      no            3.4  upsloping                  2   
301                     yes            1.2  upsloping                  1   
302                      no            0.0  upsloping                  1   

           thalassemia  target  
0               normal       1  
1         fixed defect       1  
2         fixed defect       1  
3         fixed defect       1  
4         fixed defect       1  
..                 ...     ...  
298  reversable defect       0  
299  reversable defect       0  
300  reversable defect       0  
301  reversable defect       0  
302       fixed defect       0  

[303 rows x 14 columns]

#### Описание признаков

**age**: The person's age in years

**sex**: The person's sex (1 = male, 0 = female)

**cp**: The chest pain experienced (Value 1: typical angina, Value 2: atypical angina, Value 3: non-anginal pain, Value 4: asymptomatic)

**restbps**: The person's resting blood pressure (mm Hg on admission to the hospital)

**chol**: The person's cholesterol measurement in mg/dl

**fbs**: The person's fasting blood sugar (> 120 mg/dl, 1 = true; 0 = false)

**restecg**: Resting electrocardiographic measurement (0 = normal, 1 = having ST-T wave abnormality, 2 = showing probable or definite left ventricular hypertrophy by Estes' criteria)

**thalach**: The person's maximum heart rate achieved

**exang**: Exercise induced angina (1 = yes; 0 = no)

**oldpeak**: ST depression induced by exercise relative to rest ('ST' relates to positions on the ECG plot. See more here)

**slope**: the slope of the peak exercise ST segment (Value 1: upsloping, Value 2: flat, Value 3: downsloping)

**ca**: The number of major vessels (0-3)

**thal**: A blood disorder called thalassemia (3 = normal; 6 = fixed defect; 7 = reversable defect)

**target**: Heart disease (0 = no, 1 = yes)

> 1. age 
> 2. sex 
> 3. chest pain type (4 values) 
> 4. resting blood pressure 
> 5. serum cholestoral in mg/dl 
> 6. fasting blood sugar > 120 mg/dl
> 7. resting electrocardiographic results (values 0,1,2)
> 8. maximum heart rate achieved 
> 9. exercise induced angina 
> 10. oldpeak = ST depression induced by exercise relative to rest 
> 11. the slope of the peak exercise ST segment 
> 12. number of major vessels (0-3) colored by flourosopy 
> 13. thal: 3 = normal; 6 = fixed defect; 7 = reversable defect

___

#### Сколько мужчин в датасете? Сколько женщен? (sex)

In [12]:
data.sex.value_counts()

male      207
female     96
Name: sex, dtype: int64

#### Какой процент мужчин в датасете? (решите в одну строчку, не используя предыдущий результат. Не забудте знак процента) 

In [29]:
data.sex.value_counts(normalize=True).mul(100).round(2).astype(str) + ' %'
# Умножила на 100, округлила до второго знака, изменила тип данных и добавила процент

male      68.32 %
female    31.68 %
Name: sex, dtype: object

#### Сколько мужчин имеют заболевание сердца? Сколько женщин имеют заболевание сердца?

In [56]:
crosstab = pd.crosstab(data.sex, data.target)
pd.DataFrame(crosstab, columns=[1]).loc[['male', 'female']]

1
sex       
male    93
female  72

In [61]:
crosstab = pd.crosstab(data.sex, data.target)
pd.DataFrame(crosstab, columns=[1]).sort_values(1, ascending = False)
# Сортировка по-другому

1
sex       
male    93
female  72

**Какую долю, от общего числа пациентов, занимают мужчины не имеющие болезнь сердца?**

In [ ]:
## Ваш код

**Сколько лет самому молодому пациенту, страдающему болезнью сердца?**

In [ ]:
## Ваш код

**Сколько лет самому возастному пациенту, у которого нет проблем с сердцем?**

In [ ]:
## Ваш код

**Сколько лет самой молодой женщине, которая страдает болезнью сердца?**

In [ ]:
## Ваш код

**Какой средний возраст женщин?**

In [ ]:
## Ваш код

**Каковы средние значения и среднеквадратичные отклонения возраста тех, кто страдают болезнью сердца?**

In [ ]:
## Ваш код

#### Правда ли, что люди не болеющие болезнью сердца имеют уровня холестерина меньше среднего? (chol) 

In [ ]:
## Ваш код

**Выведите статистику rest_ecg для все числовых признаков, его максимально и среднее значение (используйте groupby(), решите в одну строчку)**

In [34]:
## Ваш код

**Посчитайте у кого уровень депрессии при физический нагрузке выше (в среднем), среди мужчин страдающих болезнью сердца или среди женщин не страдающих болезнью сердца (st_depression)**

In [ ]:
## Ваш код

**Посчтитайте максимальный и минимальный уровень холестерина для каждого типа chest_pain_type, rest_ecg, thalassemia. Пишите код оптимально, можно использовать циклы**

In [ ]:
## Ваш код

**Сколько значений может принимать каждый из категориальных признаков?**

In [ ]:
## Ваш код

**У какого категориального признака наблюдается самый сильный дизбаланс классов?**

In [40]:
## Ваш код

---